src/status_im/data_store/realm/core.cljs

```diff
@@ -51,14 +51,29 @@
   (when realm
     (.close realm)))
 
-(defn migrate-realm [file-name schemas encryption-key]
-  (let [current-version (realm-version file-name encryption-key)]
-    (doseq [schema schemas
-            :when (> (:schemaVersion schema) current-version)
-            :let [migrated-realm (open-realm schema file-name encryption-key)]]
-      (close migrated-realm)))
+(defn reset-realm [file-name schemas encryption-key]
+  (utils/show-popup "Important: Wallet Upgrade" "The Status Wallet will be upgraded in this release. The 12 mnemonic words will generate different addresses and whisper identities (public key). Given that we changed the algorithm used to generate keys and addresses, it will be impossible to re-import accounts created with the old algorithm in Status. Please create a new account.")
+  (delete-realm file-name)
   (open-realm (last schemas) file-name encryption-key))
 
+(defn- migrate-schemas [file-name schemas encryption-key current-version]
+  (doseq [schema schemas
+          :when (> (:schemaVersion schema) current-version)
+          :let [migrated-realm (open-realm schema file-name encryption-key)]]
+    (close migrated-realm))
+  (open-realm (last schemas) file-name encryption-key))
+
+(defn migrate-realm [file-name schemas encryption-key]
+  (let [encrypted-version (encrypted-realm-version file-name encryption-key)
+        ;; If it's unencrypted reset schema
+        unencrypted?      (and (not encrypted-version)
+                               (unencrypted-realm? file-name))]
```

Comment:
What will happen if this is fresh install without any previous schema - won't `unencrypted?` be true ? 
If yes, will it work fine, or crash on the `reset-realm` schema (+ show confusing pop-up) ?

Reply:
that's interesting, it seems like you are right, I had tested it on my local android and worked fine, but will make sure it works.

Reply:
Weirdly enough it just works, I can't really say why from the code (possibly because opening an empty realm will succede and return a version of `0`? ). But will make sure that is the case ,good shout.